tcpdump.c

```diff
@@ -1371,8 +1405,29 @@ main(int argc, char **argv)
 #endif /* _WIN32 */
 #ifdef HAVE_PCAP_CREATE
 		pd = pcap_create(device, ebuf);
-		if (pd == NULL)
+		if (pd == NULL) {
```

Comment:
You are assuming here that `pcap_create()` will fail if an interface name is supplied that does not correspond to an interface on the system.  It is not guaranteed to do so on all platforms.  You have to wait for `pcap_activate()` to fail with `PCAP_ERROR_NO_SUCH_DEVICE` (other errors mean there *is* such a device but you can't open it).

This also doesn't handle the case of tcpdump with an older version of libpcap without `pcap_create()`.